### PR TITLE
bpo-39244: multiprocessing get all start methods

### PR DESCRIPTION
**Pull request by idomic**
# [bpo-39244](https://bugs.python.org/issue39244): multiprocessing get all start methods

# [3.8] [bpo-39244](https://bugs.python.org/issue39244): multiprocessing get all start methods (GH-18529)
# [3.9] [bpo-39244](https://bugs.python.org/issue39244): multiprocessing get all start methods (GH-18529)

https://bugs.python.org/issue39244

### CODE REVIEW

File: Lib/multiprocessing/context.py [hunk 1]
```diff
@@ -254,7 +254,7 @@ def get_start_method(self, allow_none=False):
         return self._actual_context._name
 
     def get_all_start_methods(self):
-        if sys.platform == 'win32':
+        if sys.platform == 'win32' or sys.platform == 'darwin':
```
Comment (pppery) [inline Lib/multiprocessing/context.py]: Presumably, non-spawn start methods still exist.

**Comment by stefanholek** [comment]
Not sure this is the correct fix, fork and forkserver still exist on macOS. I would think get_all_start_methods should return ['spawn', 'fork', 'forkserver'] on macOS.

File: Lib/multiprocessing/context.py (hunk 1 above)
Comment (idomic) [reply Lib/multiprocessing/context.py]: You're right! I've added a new switch case to deal only with macos

File: Lib/multiprocessing/context.py [hunk 2]
```diff
@@ -256,6 +256,8 @@ def get_start_method(self, allow_none=False):
     def get_all_start_methods(self):
         if sys.platform == 'win32':
             return ['spawn']
+        elif sys.platform == 'darwin':
+            return ['spawn', 'fork', 'forkserver']
```
Comment (pppery) [inline Lib/multiprocessing/context.py:260]: Is HAVE_SEND_HANDLE guaranteed to be defined on macOS?

Comment (idomic) [reply Lib/multiprocessing/context.py:260]: I think it's always set from a certain version (could not find documentation for that), the example of the code that defaults to this case and to 'fork' is in the bugs.python.

**Comment by idomic** [comment]
Adding @taleinat

**Comment by idomic** [comment]
This is an old PR, please refer to the clean one: GH-18625

**Pull request closed without merging**